apps/game_backend/lib/game_backend/units/skills/effects/target_strategy.ex

```diff
@@ -0,0 +1,69 @@
+defmodule GameBackend.Units.Skills.Effects.TargetStrategy do
+  @moduledoc """
+  The TargetStrategy type for skills.
+  """
+
+  use Ecto.Type
+
+  def type(), do: :string
+
+  def cast("random"), do: {:ok, "random"}
+  def cast("nearest"), do: {:ok, "nearest"}
+  def cast("furthest"), do: {:ok, "furthest"}
+  def cast("frontline"), do: {:ok, "frontline"}
+  def cast("backline"), do: {:ok, "backline"}
+  def cast(%{"factions" => factions}), do: {:ok, {"factions", factions}}
+  def cast(%{"classes" => classes}), do: {:ok, {"classes", classes}}
+  def cast(%{"lowest" => attribute}), do: {:ok, {"lowest", attribute}}
+  def cast(%{"highest" => attribute}), do: {:ok, {"highest", attribute}}
+  def cast(_), do: :error
+
+  def load(string), do: {:ok, target_strategy_from_string(string)}
+
+  def dump(strategy), do: {:ok, target_strategy_to_string(strategy)}
+
+  defp target_strategy_to_string("random"), do: "random"
+  defp target_strategy_to_string("nearest"), do: "nearest"
+  defp target_strategy_to_string("furthest"), do: "furthest"
+  defp target_strategy_to_string("frontline"), do: "frontline"
+  defp target_strategy_to_string("backline"), do: "backline"
```

Comment:
We can replace this with a `when is_binary(var)` guard

Reply:
We wont be doing this because, even tho we're validating that the strategy has some of these values, we want to check that in both ways, enconding and decoding